Source/JavaScriptCore/jit/CCallHelpers.h

```diff
@@ -531,15 +531,13 @@ class CCallHelpers : public AssemblyHelpers {
 #endif // USE(JSVALUE64)
 
     template<typename OperationType, unsigned numGPRArgs, unsigned numGPRSources, unsigned numFPRArgs, unsigned numFPRSources, unsigned numCrossSources, unsigned extraGPRArgs, unsigned nonArgGPRs, unsigned extraPoke, typename Arg, typename... Args>
-    ALWAYS_INLINE std::enable_if_t<
-        std::is_base_of<TrustedImm, Arg>::value
-        || std::is_convertible<Arg, TrustedImm>::value> // We have this since DFGSpeculativeJIT has it's own implementation of TrustedImmPtr
-    setupArgumentsImpl(ArgCollection<numGPRArgs, numGPRSources, numFPRArgs, numFPRSources, numCrossSources, extraGPRArgs, nonArgGPRs, extraPoke> argSourceRegs, Arg arg, Args... args)
+        requires WTF::DerivedFromOrConvertibleTo<Arg, TrustedImm> // We have this since DFGSpeculativeJIT has it's own implementation of TrustedImmPtr
```

Comment:
```suggestion
        requires WTF::DerivedFromOrConvertibleTo<Arg, TrustedImm> // DerivedFromOrConvertibleTo instead of derived_from since DFGSpeculativeJIT has its own implementation of TrustedImmPtr